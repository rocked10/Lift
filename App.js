import 'react-native-gesture-handler';
import React, { useState, useEffect } from 'react';
import { StyleSheet, Alert } from 'react-native';
import Login from "./screens/login";
import { TabStack } from "./routes/tabStack";
import firebase from 'firebase';
import firebaseApp from "./api/firebase";
import { MenuProvider } from 'react-native-popup-menu';
// import * as Font from 'expo-font'

export default function App() {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [loggedIn, setLoggedIn] = useState(false);

  const userDetails = (details) => {
    setEmail(details.email.toLowerCase());
    setPassword(details.password);

    firebase.auth().signInWithEmailAndPassword(email.trim(), password)
        .then((userCredential) => {
          // Signed in
          const user = userCredential.user;
          setLoggedIn(true);
        })
        .catch((error) => {
          console.log(error.message);
          Alert.alert(
              "Error",
              error.message,
              [{ text: "OK",
                onPress: () => console.log("OK Pressed") }]
          )
        });
  }

  firebase.auth().onAuthStateChanged(user => {
    if (user) {
      setLoggedIn(true);
    } else {
      setLoggedIn(false);
    }
  })

  if (loggedIn) {
    return (
      <MenuProvider>
        <TabStack />
      </MenuProvider>
    );
  } else {
    return (
        <Login userDetails={userDetails}/>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
