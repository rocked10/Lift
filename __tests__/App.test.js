import React from 'react';
import renderer from 'react-test-renderer';
import { act } from "react-native-testing-library";
import { render, fireEvent, waitFor } from '@testing-library/react-native';

import App from '../App';

describe('<App />', () => {
    it('App should render correctly', async () => {
        await act(async() => {
            const tree = renderer.create(<App />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    it('Renders Loading Screen if login status is not determined', async () => {
        const { getByTestId } = render(<App />);
        await waitFor(() => getByTestId('Loading Indicator'));
    });

    it('Renders Workout Screen if logged in', async () => { // This test also ensures TabStack returns the correct stack, and tests that WorkoutStack returns the correct screen
        const { getByTestId } = render(<App testLogin='1'/> );
        await waitFor(() => getByTestId('Add Workout')); // looks for add workout button
    });

    it('Renders Login Screen if not logged in', async () => {
        const { getByTestId } = render(<App testLogin='0'/> );
        await waitFor(() => getByTestId('Login Button')); // looks for login button
    });
});