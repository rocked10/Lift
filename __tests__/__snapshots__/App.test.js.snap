// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App should render correctly 1`] = `null`;
